src/AutoMapper/MapperConfiguration.cs

```diff
@@ -37,7 +37,8 @@ public MapperConfiguration(MapperConfigurationExpression configurationExpression
             Configuration = new ProfileMap(configurationExpression);
             Profiles = new[] { Configuration }.Concat(configurationExpression.Profiles.Select(p => new ProfileMap(p, configurationExpression))).ToArray();
 
-            configurationExpression.Advanced.BeforeSeal?.Invoke(this);
+            foreach (var beforeSealAction in configurationExpression.Advanced.BeforeSealActions)
+                beforeSealAction?.Invoke(this);
```

Comment:
Everything else goes through a foreach loop of items in this class.  Like profiles you can see the property but to add you call add function, so what's the difference?  It's exposed as an enumerable as to not add another way.  I don't see changing it as "that" big of a deal in exposing it or not.